src/olympia/scanners/tasks.py

```diff
@@ -182,6 +182,10 @@ def run_yara(results, upload_pk):
        results as first argument.
     - `upload_pk` is the FileUpload ID.
     """
+    return _run_yara(results, upload_pk)
+
+
+def _run_yara(results, upload_pk):
```

Comment:
What does this change (moving all the code into a separate function) do? Does the function where the exception is raised need to be *not* a `@validation_task`?

Reply:
Oh, after looking at the tests I think this is just to support the tests, or maybe not?

Reply:
yes, exactly. `@validation_task` catches all exceptions so the new test cases wouldn't show that we actually raise exceptions there.